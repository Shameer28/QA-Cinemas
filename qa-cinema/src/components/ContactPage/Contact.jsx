import React, { useState } from "react";
import "bootstrap/dist/css/bootstrap.min.css";

import Container from "react-bootstrap/Container";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import Jumbotron from "react-bootstrap/Jumbotron";
import Alert from "react-bootstrap/Alert";
import GoogleMaps from "../GettingTherePage/GoogleMaps";


const Contact = () => {

  const [showMessage, setShowMessage] = useState(false);

  const [Email, setEmail] = useState('');
  const [FirstName, setFirstName] = useState('');
  const [LastName, setLastName] = useState('');
  const [TextArea, setTextArea] = useState('');



  const handleSubmit = (e) => {
    //Preventing the page from refreshing
    e.preventDefault();

    const dataToSend = JSON.stringify({ Email, FirstName, LastName, TextArea });
    console.log("data has been sent");
    console.log(dataToSend);


    //Setting the states back to ''
    setEmail('');
    setFirstName('');
    setLastName('');
    setTextArea('');


    // const handleSubmits = (e) => {
    //     e.preventDefault();
    //     axios({
    //         method: "POST",
    //         url: "http://localhost:3000/send",
    //         data: this.data
    //     }).then((response) => {
    //         if (response.data.status === 'success') {
    //             alert("Message Sent.");
    //             this.resetForm()
    //         } else if (response.data.status === 'fail') {
    //             alert("Message failed to send.")
    //         }
    //     });
    // };
  }

  return (
    <div>
      <div >
        <Container className="p-3">
          <Container>
            <Row>
              <Col>
                <Alert show={showMessage} variant="success">
                  <Alert.Heading>Success</Alert.Heading>
                  <p>
                    This is an alert, and these can be shown in a variety of
                    styles provided by Bootstrap. Our form submit button simply
                    shows this alert. Another post will go through form validation
                    and errors.
                  </p>
                  <hr />
                </Alert>
              </Col>
            </Row>
            <Row>
              <Col>
                <Jumbotron>
                  <GoogleMaps />
                </Jumbotron>
              </Col>

              <Col style={{ margin: "auto", width: "50%" }}>
                <div style={{ margin: "auto", width: "50%" }}>
                  <div class="text"><h2>Feel free to get in contact with us:</h2> </div>
                  <br></br>
                  <div>
                    <strong>Location: </strong>
                    8, St James's Bldg, 61-95 Oxford St, Manchester, M1 6FQ
                  </div>
                  <br></br>
                  <div>
                    <strong>Call Center: </strong>
                    <a href="tel:+44 (0)345 074 7995">+44 (0)345 074 7995</a>
                  </div>
                  <br></br>
                  <div>
                    <strong>Email Us: </strong>
                    <a href="#">SDar@qa.com</a>
                  </div>
                </div>
              </Col>
            </Row>
          </Container>
        </Container>
      </div>
    </div>
  );
}

export default Contact;