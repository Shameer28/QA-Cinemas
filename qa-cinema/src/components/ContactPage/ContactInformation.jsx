
import Container from 'react-bootstrap/Container';

const ContactInformation = () => {
    return (
        <Container>

            <div class="column col-lg-4 col-md-12 col-sm-12">
                <div class="text"><h2>Feel free to get in contact with us.</h2> </div>
                <ul class="contact-info">
                    <li>
                        <strong>Location: </strong>
                        8, St James's Bldg, 61-95 Oxford St, Manchester, M1 6FQ
                    </li>
                    <li>

                        <strong>Call Center: </strong>
                        <a href="tel:+44 (0)345 074 7995">+44 (0)345 074 7995</a>
                    </li>
                    <li>
                        <strong>Email Us: </strong>
                        <a href="#">SDar@qa.com</a>
                    </li>

                </ul>
            </div>
        </Container>
    );
}

export default ContactInformation;