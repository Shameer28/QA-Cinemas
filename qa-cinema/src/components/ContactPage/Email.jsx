import React, { useState } from "react";
import "bootstrap/dist/css/bootstrap.min.css";
import Card from 'react-bootstrap/Card'
import Container from "react-bootstrap/Container";
import Form from "react-bootstrap/Form";
import { Button } from "react-bootstrap";

const Email = () => {
    const [Email, setEmail] = useState('');
    const [FirstName, setFirstName] = useState('');
    const [LastName, setLastName] = useState('');
    const [TextArea, setTextArea] = useState('');




    const handleSubmit = (e) => {
        //Preventing the page from refreshing
        e.preventDefault();

        const dataToSend = JSON.stringify({ Email, FirstName, LastName, TextArea });
        console.log("data has been sent");
            

        //Setting the states back to ''
        setEmail('');
        setFirstName('');
        setLastName('');
        setTextArea('');


        // const handleSubmits = (e) => {
        //     e.preventDefault();
        //     axios({
        //         method: "POST",
        //         url: "http://localhost:3000/send",
        //         data: this.data
        //     }).then((response) => {
        //         if (response.data.status === 'success') {
        //             alert("Message Sent.");
        //             this.resetForm()
        //         } else if (response.data.status === 'fail') {
        //             alert("Message failed to send.")
        //         }
        //     });
        // };

    }
    return (
        <div style={{ width: "100%", backgroundColor: "#212529", color: "white" }}>
            <Container>
                <Card className="text-center" style={{ backgroundColor: "#212529", color: "white" }}>
                    <Card.Header>Contact US</Card.Header>
                    <Card.Body>
                        <Card.Title><u>Please fill out the form to submit a request</u></Card.Title>
                        <Card.Text>
                            <Form onSubmit={handleSubmit} >
                                <Form.Group>
                                    <Form.Label>Email address</Form.Label>
                                    <Form.Control id="Email" type="email" className="form-control" placeholder="name@example.com" value={Email} required onChange={(e) => { setEmail(e.target.value) }} />
                                </Form.Group>
                                <Form.Group >
                                    <Form.Label>First Name</Form.Label>
                                    <Form.Control id="FirstName" type="text" className="form-control" required minLength={3} placeholder="First Name" value={FirstName} onChange={(e) => { setFirstName(e.target.value) }} />

                                    <Form.Label>Last Name</Form.Label>
                                    <Form.Control id="LastName" type="text" className="form-control" required minLength={3} placeholder="Last Name" value={LastName} onChange={(e) => { setLastName(e.target.value) }} />
                                </Form.Group>
                                <Form.Group >
                                    <Form.Label>Message</Form.Label>
                                    <Form.Control id="TextArea" as="textarea" className="form-control" required minLength={3} rows={3} value={TextArea} onChange={(e) => { setTextArea(e.target.value) }} />
                                </Form.Group>
                                <br />
                                <Button type="submit"> Submit</Button>
                            </Form>
                        </Card.Text>

                    </Card.Body>
                    <Card.Footer className="text-muted"></Card.Footer>
                </Card>
            </Container>
        </div>
    );
}
export default Email;