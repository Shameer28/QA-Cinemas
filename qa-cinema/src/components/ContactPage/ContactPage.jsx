import Contact from "./Contact";
import Email from "./Email";
const ContactPage = () => {
    return (
        <div class="background">
            <Contact />
            <Email/>
            
        </div>
    );
}

export default ContactPage;