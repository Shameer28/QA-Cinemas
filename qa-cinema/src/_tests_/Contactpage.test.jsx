import ContactPage from '../components/ContactPage/ContactPage'
import { create } from 'react-test-renderer';


describe(`Contact Page testing`, () => {

    it(`Should match the snapshot`, () => {
        const instance = create(<ContactPage/>).toJSON();
        expect(instance).toMatchSnapshot();
    })
})