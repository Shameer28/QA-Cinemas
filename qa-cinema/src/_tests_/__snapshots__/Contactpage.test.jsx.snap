// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Page testing Should match the snapshot 1`] = `
<div
  class="background"
>
  <img
    alt="ContactUS"
    fluid={true}
    src="contact-us-banner.png"
    width="100%"
  />
  <br />
  <br />
  <div
    style={
      Object {
        "backgroundColor": "white",
        "border": "solid",
        "marginLeft": "auto",
        "marginRight": "auto",
        "width": "50%",
      }
    }
  >
    <div>
      <div>
        <div
          className="p-3 container"
        >
          <div
            className="container"
          >
            <div
              className="row"
            >
              <div
                className="col"
              >
                <div
                  className="jumbotron"
                >
                  <div
                    style={
                      Object {
                        "border": "solid",
                      }
                    }
                  >
                    <div
                      style={
                        Object {
                          "height": "600px",
                          "width": "100%",
                        }
                      }
                    >
                      <div
                        onClick={[Function]}
                        onMouseDownCapture={[Function]}
                        onMouseMove={[Function]}
                        style={
                          Object {
                            "height": "100%",
                            "margin": 0,
                            "padding": 0,
                            "position": "relative",
                            "width": "100%",
                          }
                        }
                      >
                        <div
                          style={
                            Object {
                              "height": "100%",
                              "left": 0,
                              "margin": 0,
                              "padding": 0,
                              "position": "absolute",
                              "top": 0,
                              "width": "100%",
                            }
                          }
                        />
                        <div
                          style={
                            Object {
                              "height": "50%",
                              "left": "50%",
                              "margin": 0,
                              "padding": 0,
                              "position": "absolute",
                              "top": "50%",
                              "width": "50%",
                            }
                          }
                        >
                          <div
                            style={
                              Object {
                                "height": "100%",
                                "left": 0,
                                "margin": 0,
                                "padding": 0,
                                "position": "absolute",
                                "top": 0,
                                "width": "100%",
                              }
                            }
                          >
                            <div
                              style={
                                Object {
                                  "backgroundColor": "transparent",
                                  "height": 0,
                                  "left": 0,
                                  "position": "absolute",
                                  "top": 0,
                                  "width": 0,
                                }
                              }
                            >
                              <div>
                                <div
                                  className="pin bounce"
                                  style={
                                    Object {
                                      "backgroundColor": "red",
                                      "cursor": "pointer",
                                    }
                                  }
                                  title="QA Training Building 1"
                                />
                                <div
                                  className="pulse"
                                />
                              </div>
                            </div>
                            <div
                              style={
                                Object {
                                  "backgroundColor": "transparent",
                                  "height": 0,
                                  "left": -4140.112093161792,
                                  "position": "absolute",
                                  "top": 124.22172901593149,
                                  "width": 0,
                                }
                              }
                            >
                              <div>
                                <div
                                  className="pin bounce"
                                  style={
                                    Object {
                                      "backgroundColor": "red",
                                      "cursor": "pointer",
                                    }
                                  }
                                  title="QA Training Building 2"
                                />
                                <div
                                  className="pulse"
                                />
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="col"
                style={
                  Object {
                    "margin": "auto",
                    "width": "50%",
                  }
                }
              >
                <div
                  style={
                    Object {
                      "margin": "auto",
                    }
                  }
                >
                  <div
                    class="text"
                  >
                    <h2>
                      <u>
                        Feel free to get in contact with us:
                      </u>
                    </h2>
                     
                  </div>
                  <br />
                  <div>
                    <strong>
                      Location: 
                    </strong>
                    8, St James's Bldg, 61-95 Oxford St, Manchester, M1 6FQ
                  </div>
                  <br />
                  <div>
                    <strong>
                      Call Center: 
                    </strong>
                    <a
                      href="tel:+44 (0)345 074 7995"
                    >
                      +44 (0)345 074 7995
                    </a>
                  </div>
                  <br />
                  <div>
                    <strong>
                      Email Us: 
                    </strong>
                    <a
                      href="mailto:SDar@qa.com"
                    >
                      SDar@qa.com
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        className="container"
      >
        <div
          className="text-center card"
        >
          <div
            className="card-header"
          >
            Contact US
          </div>
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
              <u>
                Please fill out the form to submit a request
              </u>
            </div>
            <p
              className="card-text"
            >
              <form
                className=""
                onSubmit={[Function]}
              >
                <div
                  className="form-group"
                >
                  <label
                    className="form-label"
                  >
                    Email address
                  </label>
                  <input
                    className="form-control form-control"
                    id="Email"
                    onChange={[Function]}
                    placeholder="name@example.com"
                    required={true}
                    type="email"
                    value=""
                  />
                </div>
                <div
                  className="form-group"
                >
                  <label
                    className="form-label"
                  >
                    First Name
                  </label>
                  <input
                    className="form-control form-control"
                    id="FirstName"
                    minLength={3}
                    onChange={[Function]}
                    placeholder="First Name"
                    required={true}
                    type="text"
                    value=""
                  />
                  <label
                    className="form-label"
                  >
                    Last Name
                  </label>
                  <input
                    className="form-control form-control"
                    id="LastName"
                    minLength={3}
                    onChange={[Function]}
                    placeholder="Last Name"
                    required={true}
                    type="text"
                    value=""
                  />
                </div>
                <div
                  className="form-group"
                >
                  <label
                    className="form-label"
                  >
                    Message
                  </label>
                  <textarea
                    className="form-control form-control"
                    id="TextArea"
                    minLength={3}
                    onChange={[Function]}
                    required={true}
                    rows={3}
                    value=""
                  />
                </div>
                <br />
                <button
                  id="submit"
                  type="submit"
                >
                   submit
                </button>
              </form>
            </p>
          </div>
          <div
            className="text-muted card-footer"
          />
        </div>
      </div>
    </div>
  </div>
  <br />
</div>
`;
